cmd/gen.go

```diff
@@ -25,6 +25,23 @@ import (
 	"github.com/istio-ecosystem/coddiwomple/pkg/routing"
 )
 
+func genCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "gen",
+		Short: "Generate config files",
+		Long:  `Generate various Istio resource descriptions, and Coddiwomple config files`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return errors.New("cannot use gen directly, use a subcommand")
+		},
+	}
+
+	cmd.AddCommand(
+		configGenCmd(),
+	)
+
+	return cmd
+}
+
 func configGenCmd() *cobra.Command {
```

Comment:
move into `cmd/gen`/`package gen`. Keep the top level `cmd/gen.go` with `func genCmd()` in it. Roughly, every top level `cw` command should be a file in `cmd/`, if they have subcommands those subcommands should live in `cmd/<subcommand>/`, and finally the top level `cw` itself lives in `cmd/cw` (since that makes naming nice and easy for us).